Q1. A company conducted a survey of its employees and found that 70% of the employees use the 
company's health insurance plan, while 40% of the employees who use the plan are smokers. What is the 
probability that an employee is a smoker given that he/she uses the health insurance plan?

Q2. What is the difference between Bernoulli Naive Bayes and Multinomial Naive Bayes?

Q3. How does Bernoulli Naive Bayes handle missing values?

Q4. Can Gaussian Naive Bayes be used for multi-class classification?

Q5. Assignment:
Data preparation:

Download the "Spambase Data Set" from the UCI Machine Learning Repository (https://archive.ics.uci.edu/ml/
datasets/Spambase). This dataset contains email messages, where the goal is to predict whether a message 
is spam or not based on several input features.


Implementation:

Implement Bernoulli Naive Bayes, Multinomial Naive Bayes, and Gaussian Naive Bayes classifiers using the 
scikit-learn library in Python. Use 10-fold cross-validation to evaluate the performance of each classifier on the 
dataset. You should use the default hyperparameters for each classifier.


Results:

Report the following performance metrics for each classifier:

Accuracy

Precision

Recall

F1 score


Discussion:

Discuss the results you obtained. Which variant of Naive Bayes performed the best? Why do you think that is 
the case? Are there any limitations of Naive Bayes that you observed?


Conclusion:

Summarise your findings and provide some suggestions for future work



- Bernoulli Naive Bayes is used for binary or boolean features, while Multinomial Naive Bayes is used for discrete features with multiple values.
- Bernoulli Naive Bayes explicitly models the presence or absence of a feature, while Multinomial Naive Bayes models the number of counts of a feature.
- Bernoulli Naive Bayes would be appropriate for classifying spam emails based on the presence or absence of certain words, while Multinomial Naive Bayes would be more appropriate for classifying documents based on the frequency of occurrence of certain words.


Bernoulli Naive Bayes is primarily designed for binary data, where features represent the presence (1) or absence (0) of specific attributes. When dealing with missing values in Bernoulli Naive Bayes

1. Ignoring Missing Values: Bernoulli Naive Bayes typically ignores missing values by treating them as if the attribute is absent (0). This is consistent with the binary nature of the algorithm, where each feature is either present (1) or absent (0). So, if a feature is missing, it's effectively considered as "not present" (0).

2. Imputation: In some cases, you might choose to impute missing values before applying Bernoulli Naive Bayes. Imputation means filling in the missing values with a specific value (either 0 or 1). The choice of imputation depends on your problem and data. For Bernoulli Naive Bayes, you can either impute with 0 (absence) or 1 (presence) based on domain knowledge or the nature of your dataset.

3. Imputing with Probabilities: In more advanced scenarios, you might use statistical techniques to impute missing values probabilistically. For example, you could calculate the probability of an attribute being present (1) based on the overall distribution of that attribute in the dataset and use that probability to impute missing values. This approach attempts to capture the uncertainty associated with imputing missing data.


Yes, Gaussian Naive Bayes can be used for multi-class classification tasks. Gaussian Naive Bayes is an extension of the Naive Bayes algorithm that is specifically designed for continuous data, and it can handle multi-class classification problems where the target variable has more than two classes.

 Gaussian Naive Bayes can handle multi-class classification problems by modeling the data distribution for each class as Gaussian and then calculating the most likely class for a given data point. It's a simple and efficient algorithm for both binary and multi-class classification tasks, although it does make the assumption of feature independence within each class, which may not always hold in practice.

In [5]:
import pandas as pd 
import numpy as np 


In [13]:
data = pd.read_csv('spambase.csv')

In [14]:
data

0  0.64  0.64.1  0.1  0.32   0.2   0.3   0.4   0.5   0.6  ...   0.41  \
0     0.21  0.28    0.50  0.0  0.14  0.28  0.21  0.07  0.00  0.94  ...  0.000   
1     0.06  0.00    0.71  0.0  1.23  0.19  0.19  0.12  0.64  0.25  ...  0.010   
2     0.00  0.00    0.00  0.0  0.63  0.00  0.31  0.63  0.31  0.63  ...  0.000   
3     0.00  0.00    0.00  0.0  0.63  0.00  0.31  0.63  0.31  0.63  ...  0.000   
4     0.00  0.00    0.00  0.0  1.85  0.00  0.00  1.85  0.00  0.00  ...  0.000   
...    ...   ...     ...  ...   ...   ...   ...   ...   ...   ...  ...    ...   
4595  0.31  0.00    0.62  0.0  0.00  0.31  0.00  0.00  0.00  0.00  ...  0.000   
4596  0.00  0.00    0.00  0.0  0.00  0.00  0.00  0.00  0.00  0.00  ...  0.000   
4597  0.30  0.00    0.30  0.0  0.00  0.00  0.00  0.00  0.00  0.00  ...  0.102   
4598  0.96  0.00    0.00  0.0  0.32  0.00  0.00  0.00  0.00  0.00  ...  0.000   
4599  0.00  0.00    0.65  0.0  0.00  0.00  0.00  0.00  0.00  0.00  ...  0.000   

       0.42  0.43  0.778   0.44   0.45  3.756   61   278  1  
0     0.132   0.0  0.372  0.180  0.048  5.114  101  1028  1  
1     0.143   0.0  0.276  0.184  0.010  9.821  485  2259  1  
2     0.137   0.0  0.137  0.000  0.000  3.537   40   191  1  
3     0.135   0.0  0.135  0.000  0.000  3.537   40   191  1  
4     0.223   0.0  0.000  0.000  0.000  3.000   15    54  1  
...     ...   ...    ...    ...    ...    ...  ...   ... ..  
4595  0.232   0.0  0.000  0.000  0.000  1.142    3    88  0  
4596  0.000   0.0  0.353  0.000  0.000  1.555    4    14  0  
4597  0.718   0.0  0.000  0.000  0.000  1.404    6   118  0  
4598  0.057   0.0  0.000  0.000  0.000  1.147    5    78  0  
4599  0.000   0.0  0.125  0.000  0.000  1.250    5    40  0  

[4600 rows x 58 columns]

In [15]:
feature_names = [
    'word_freq_make', 'word_freq_address', 'word_freq_all', 'word_freq_3d',
    'word_freq_our', 'word_freq_over', 'word_freq_remove', 'word_freq_internet',
    'word_freq_order', 'word_freq_mail', 'word_freq_receive', 'word_freq_will',
    'word_freq_people', 'word_freq_report', 'word_freq_addresses', 'word_freq_free',
    'word_freq_business', 'word_freq_email', 'word_freq_you', 'word_freq_credit',
    'word_freq_your', 'word_freq_font', 'word_freq_000', 'word_freq_money',
    'word_freq_hp', 'word_freq_hpl', 'word_freq_george', 'word_freq_650',
    'word_freq_lab', 'word_freq_labs', 'word_freq_telnet', 'word_freq_857',
    'word_freq_data', 'word_freq_415', 'word_freq_85', 'word_freq_technology',
    'word_freq_1999', 'word_freq_parts', 'word_freq_pm', 'word_freq_direct',
    'word_freq_cs', 'word_freq_meeting', 'word_freq_original', 'word_freq_project',
    'word_freq_re', 'word_freq_edu', 'word_freq_table', 'word_freq_conference',
    'char_freq_;', 'char_freq_(', 'char_freq_[', 'char_freq_!', 'char_freq_$',
    'char_freq_#', 'capital_run_length_average', 'capital_run_length_longest',
    'capital_run_length_total' , 'output'
]


In [16]:
data = pd.DataFrame([feature_names] + data.values.tolist(), columns=feature_names)

In [17]:
data

word_freq_make  word_freq_address  word_freq_all  word_freq_3d  \
0     word_freq_make  word_freq_address  word_freq_all  word_freq_3d   
1               0.21               0.28            0.5           0.0   
2               0.06                0.0           0.71           0.0   
3                0.0                0.0            0.0           0.0   
4                0.0                0.0            0.0           0.0   
...              ...                ...            ...           ...   
4596            0.31                0.0           0.62           0.0   
4597             0.0                0.0            0.0           0.0   
4598             0.3                0.0            0.3           0.0   
4599            0.96                0.0            0.0           0.0   
4600             0.0                0.0           0.65           0.0   

      word_freq_our  word_freq_over  word_freq_remove  word_freq_internet  \
0     word_freq_our  word_freq_over  word_freq_remove  word_freq_internet   
1              0.14            0.28              0.21                0.07   
2              1.23            0.19              0.19                0.12   
3              0.63             0.0              0.31                0.63   
4              0.63             0.0              0.31                0.63   
...             ...             ...               ...                 ...   
4596            0.0            0.31               0.0                 0.0   
4597            0.0             0.0               0.0                 0.0   
4598            0.0             0.0               0.0                 0.0   
4599           0.32             0.0               0.0                 0.0   
4600            0.0             0.0               0.0                 0.0   

      word_freq_order  word_freq_mail  ...  char_freq_;  char_freq_(  \
0     word_freq_order  word_freq_mail  ...  char_freq_;  char_freq_(   
1                 0.0            0.94  ...          0.0        0.132   
2                0.64            0.25  ...         0.01        0.143   
3                0.31            0.63  ...          0.0        0.137   
4                0.31            0.63  ...          0.0        0.135   
...               ...             ...  ...          ...          ...   
4596              0.0             0.0  ...          0.0        0.232   
4597              0.0             0.0  ...          0.0          0.0   
4598              0.0             0.0  ...        0.102        0.718   
4599              0.0             0.0  ...          0.0        0.057   
4600              0.0             0.0  ...          0.0          0.0   

      char_freq_[  char_freq_!  char_freq_$  char_freq_#  \
0     char_freq_[  char_freq_!  char_freq_$  char_freq_#   
1             0.0        0.372         0.18        0.048   
2             0.0        0.276        0.184         0.01   
3             0.0        0.137          0.0          0.0   
4             0.0        0.135          0.0          0.0   
...           ...          ...          ...          ...   
4596          0.0          0.0          0.0          0.0   
4597          0.0        0.353          0.0          0.0   
4598          0.0          0.0          0.0          0.0   
4599          0.0          0.0          0.0          0.0   
4600          0.0        0.125          0.0          0.0   

      capital_run_length_average  capital_run_length_longest  \
0     capital_run_length_average  capital_run_length_longest   
1                          5.114                       101.0   
2                          9.821                       485.0   
3                          3.537                        40.0   
4                          3.537                        40.0   
...                          ...                         ...   
4596                       1.142                         3.0   
4597                       1.555                         4.0   
4598                       1.404                      

In [19]:
data = data.drop(0)

In [23]:
data.to_csv('modifiedspamdata.csv' , index=False)

In [33]:
## reading the new data

In [25]:
df = pd.read_csv('modifiedspamdata.csv')

In [26]:
df

word_freq_make  word_freq_address  word_freq_all  word_freq_3d  \
0               0.21               0.28           0.50           0.0   
1               0.06               0.00           0.71           0.0   
2               0.00               0.00           0.00           0.0   
3               0.00               0.00           0.00           0.0   
4               0.00               0.00           0.00           0.0   
...              ...                ...            ...           ...   
4595            0.31               0.00           0.62           0.0   
4596            0.00               0.00           0.00           0.0   
4597            0.30               0.00           0.30           0.0   
4598            0.96               0.00           0.00           0.0   
4599            0.00               0.00           0.65           0.0   

      word_freq_our  word_freq_over  word_freq_remove  word_freq_internet  \
0              0.14            0.28              0.21                0.07   
1              1.23            0.19              0.19                0.12   
2              0.63            0.00              0.31                0.63   
3              0.63            0.00              0.31                0.63   
4              1.85            0.00              0.00                1.85   
...             ...             ...               ...                 ...   
4595           0.00            0.31              0.00                0.00   
4596           0.00            0.00              0.00                0.00   
4597           0.00            0.00              0.00                0.00   
4598           0.32            0.00              0.00                0.00   
4599           0.00            0.00              0.00                0.00   

      word_freq_order  word_freq_mail  ...  char_freq_;  char_freq_(  \
0                0.00            0.94  ...        0.000        0.132   
1                0.64            0.25  ...        0.010        0.143   
2                0.31            0.63  ...        0.000        0.137   
3                0.31            0.63  ...        0.000        0.135   
4                0.00            0.00  ...        0.000        0.223   
...               ...             ...  ...          ...          ...   
4595             0.00            0.00  ...        0.000        0.232   
4596             0.00            0.00  ...        0.000        0.000   
4597             0.00            0.00  ...        0.102        0.718   
4598             0.00            0.00  ...        0.000        0.057   
4599             0.00            0.00  ...        0.000        0.000   

      char_freq_[  char_freq_!  char_freq_$  char_freq_#  \
0             0.0        0.372        0.180        0.048   
1             0.0        0.276        0.184        0.010   
2             0.0        0.137        0.000        0.000   
3             0.0        0.135        0.000        0.000   
4             0.0        0.000        0.000        0.000   
...           ...          ...          ...          ...   
4595          0.0        0.000        0.000        0.000   
4596          0.0        0.353        0.000        0.000   
4597          0.0        0.000        0.000        0.000   
4598          0.0        0.000        0.000        0.000   
4599          0.0        0.125        0.000        0.000   

      capital_run_length_average  capital_run_length_longest  \
0                          5.114                       101.0   
1                          9.821                       485.0   
2                          3.537                        40.0   
3                          3.537                        40.0   
4                          3.000                        15.0   
...                          ...                         ...   
4595                       1.142                         3.0   
4596                       1.555                         4.0   
4597                       1.404                      

In [27]:
df.head()

word_freq_make  word_freq_address  word_freq_all  word_freq_3d  \
0            0.21               0.28           0.50           0.0   
1            0.06               0.00           0.71           0.0   
2            0.00               0.00           0.00           0.0   
3            0.00               0.00           0.00           0.0   
4            0.00               0.00           0.00           0.0   

   word_freq_our  word_freq_over  word_freq_remove  word_freq_internet  \
0           0.14            0.28              0.21                0.07   
1           1.23            0.19              0.19                0.12   
2           0.63            0.00              0.31                0.63   
3           0.63            0.00              0.31                0.63   
4           1.85            0.00              0.00                1.85   

   word_freq_order  word_freq_mail  ...  char_freq_;  char_freq_(  \
0             0.00            0.94  ...         0.00        0.132   
1             0.64            0.25  ...         0.01        0.143   
2             0.31            0.63  ...         0.00        0.137   
3             0.31            0.63  ...         0.00        0.135   
4             0.00            0.00  ...         0.00        0.223   

   char_freq_[  char_freq_!  char_freq_$  char_freq_#  \
0          0.0        0.372        0.180        0.048   
1          0.0        0.276        0.184        0.010   
2          0.0        0.137        0.000        0.000   
3          0.0        0.135        0.000        0.000   
4          0.0        0.000        0.000        0.000   

   capital_run_length_average  capital_run_length_longest  \
0                       5.114                       101.0   
1                       9.821                       485.0   
2                       3.537                        40.0   
3                       3.537                        40.0   
4                       3.000                        15.0   

   capital_run_length_total  output  
0                    1028.0     1.0  
1                    2259.0     1.0  
2                     191.0     1.0  
3                     191.0     1.0  
4                      54.0     1.0  

[5 rows x 58 columns]

In [28]:
df.describe()

word_freq_make  word_freq_address  word_freq_all  word_freq_3d  \
count     4600.000000        4600.000000    4600.000000   4600.000000   
mean         0.104576           0.212922       0.280578      0.065439   
std          0.305387           1.290700       0.504170      1.395303   
min          0.000000           0.000000       0.000000      0.000000   
25%          0.000000           0.000000       0.000000      0.000000   
50%          0.000000           0.000000       0.000000      0.000000   
75%          0.000000           0.000000       0.420000      0.000000   
max          4.540000          14.280000       5.100000     42.810000   

       word_freq_our  word_freq_over  word_freq_remove  word_freq_internet  \
count    4600.000000     4600.000000       4600.000000         4600.000000   
mean        0.312222        0.095922          0.114233            0.105317   
std         0.672586        0.273850          0.391480            0.401112   
min         0.000000        0.000000          0.000000            0.000000   
25%         0.000000        0.000000          0.000000            0.000000   
50%         0.000000        0.000000          0.000000            0.000000   
75%         0.382500        0.000000          0.000000            0.000000   
max        10.000000        5.880000          7.270000           11.110000   

       word_freq_order  word_freq_mail  ...  char_freq_;  char_freq_(  \
count      4600.000000     4600.000000  ...  4600.000000  4600.000000   
mean          0.090087        0.239465  ...     0.038583     0.139061   
std           0.278643        0.644816  ...     0.243497     0.270377   
min           0.000000        0.000000  ...     0.000000     0.000000   
25%           0.000000        0.000000  ...     0.000000     0.000000   
50%           0.000000        0.000000  ...     0.000000     0.065000   
75%           0.000000        0.160000  ...     0.000000     0.188000   
max           5.260000       18.180000  ...     4.385000     9.752000   

       char_freq_[  char_freq_!  char_freq_$  char_freq_#  \
count  4600.000000  4600.000000  4600.000000  4600.000000   
mean      0.016980     0.268960     0.075827     0.044248   
std       0.109406     0.815726     0.245906     0.429388   
min       0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.314250     0.052000     0.000000   
max       4.081000    32.478000     6.003000    19.829000   

       capital_run_length_average  capital_run_length_longest  \
count                 4600.000000                 4600.000000   
mean                     5.191827                   52.170870   
std                     31.732891                  194.912453   
min                      1.000000                    1.000000   
25%                      1.588000                    6.000000   
50%                      2.275500                   15.000000   
75%                      3.705250                   43.000000   
max                   1102.500000                 9989.000000   

       capital_run_length_total       output  
count               4600.000000  4600.000000  
mean                 283.290435     0.393913  
std                  606.413764     0.488669  
min                    1.000000     0.000000  
25%                   35.000000     0.000000  
50%                   95.000000     0.000000  
75%                  265.250000     1.000000  
max                15841.000000     1.000000  

[8 rows x 58 columns]

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4600 entries, 0 to 4599
Data columns (total 58 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   word_freq_make              4600 non-null   float64
 1   word_freq_address           4600 non-null   float64
 2   word_freq_all               4600 non-null   float64
 3   word_freq_3d                4600 non-null   float64
 4   word_freq_our               4600 non-null   float64
 5   word_freq_over              4600 non-null   float64
 6   word_freq_remove            4600 non-null   float64
 7   word_freq_internet          4600 non-null   float64
 8   word_freq_order             4600 non-null   float64
 9   word_freq_mail              4600 non-null   float64
 10  word_freq_receive           4600 non-null   float64
 11  word_freq_will              4600 non-null   float64
 12  word_freq_people            4600 non-null   float64
 13  word_freq_report            4600 

In [31]:
df.isnull().sum()

word_freq_make                0
word_freq_address             0
word_freq_all                 0
word_freq_3d                  0
word_freq_our                 0
word_freq_over                0
word_freq_remove              0
word_freq_internet            0
word_freq_order               0
word_freq_mail                0
word_freq_receive             0
word_freq_will                0
word_freq_people              0
word_freq_report              0
word_freq_addresses           0
word_freq_free                0
word_freq_business            0
word_freq_email               0
word_freq_you                 0
word_freq_credit              0
word_freq_your                0
word_freq_font                0
word_freq_000                 0
word_freq_money               0
word_freq_hp                  0
word_freq_hpl                 0
word_freq_george              0
word_freq_650                 0
word_freq_lab                 0
word_freq_labs                0
word_freq_telnet              0
word_fre

In [32]:
# there are no null values !

In [35]:
# let us train the model

In [36]:
X = df.iloc[: , : -1]

In [37]:
X

word_freq_make  word_freq_address  word_freq_all  word_freq_3d  \
0               0.21               0.28           0.50           0.0   
1               0.06               0.00           0.71           0.0   
2               0.00               0.00           0.00           0.0   
3               0.00               0.00           0.00           0.0   
4               0.00               0.00           0.00           0.0   
...              ...                ...            ...           ...   
4595            0.31               0.00           0.62           0.0   
4596            0.00               0.00           0.00           0.0   
4597            0.30               0.00           0.30           0.0   
4598            0.96               0.00           0.00           0.0   
4599            0.00               0.00           0.65           0.0   

      word_freq_our  word_freq_over  word_freq_remove  word_freq_internet  \
0              0.14            0.28              0.21                0.07   
1              1.23            0.19              0.19                0.12   
2              0.63            0.00              0.31                0.63   
3              0.63            0.00              0.31                0.63   
4              1.85            0.00              0.00                1.85   
...             ...             ...               ...                 ...   
4595           0.00            0.31              0.00                0.00   
4596           0.00            0.00              0.00                0.00   
4597           0.00            0.00              0.00                0.00   
4598           0.32            0.00              0.00                0.00   
4599           0.00            0.00              0.00                0.00   

      word_freq_order  word_freq_mail  ...  word_freq_conference  char_freq_;  \
0                0.00            0.94  ...                   0.0        0.000   
1                0.64            0.25  ...                   0.0        0.010   
2                0.31            0.63  ...                   0.0        0.000   
3                0.31            0.63  ...                   0.0        0.000   
4                0.00            0.00  ...                   0.0        0.000   
...               ...             ...  ...                   ...          ...   
4595             0.00            0.00  ...                   0.0        0.000   
4596             0.00            0.00  ...                   0.0        0.000   
4597             0.00            0.00  ...                   0.0        0.102   
4598             0.00            0.00  ...                   0.0        0.000   
4599             0.00            0.00  ...                   0.0        0.000   

      char_freq_(  char_freq_[  char_freq_!  char_freq_$  char_freq_#  \
0           0.132          0.0        0.372        0.180        0.048   
1           0.143          0.0        0.276        0.184        0.010   
2           0.137          0.0        0.137        0.000        0.000   
3           0.135          0.0        0.135        0.000        0.000   
4           0.223          0.0        0.000        0.000        0.000   
...           ...          ...          ...          ...          ...   
4595        0.232          0.0        0.000        0.000        0.000   
4596        0.000          0.0        0.353        0.000        0.000   
4597        0.718          0.0        0.000        0.000        0.000   
4598        0.057          0.0        0.000        0.000        0.000   
4599        0.000          0.0        0.125        0.000        0.000   

      capital_run_length_average  capital_run_length_longest  \
0                          5.114                       101.0   
1                          9.821                       485.0   
2                          3.537                        40.0   
3                          3.537                        40.0   
4                          3.000              

In [38]:
y = df['output']

In [39]:
y

0       1.0
1       1.0
2       1.0
3       1.0
4       1.0
       ... 
4595    0.0
4596    0.0
4597    0.0
4598    0.0
4599    0.0
Name: output, Length: 4600, dtype: float64

In [40]:
# we sorted out dependent and independent features . 


In [41]:
from sklearn.model_selection import train_test_split
X_train , X_test , y_train , y_test = train_test_split(X,y,test_size=0.3 , random_state=0 )

# GUASIAN NAIVE BAYES CLASSIFIER

In [42]:
from sklearn.naive_bayes import GaussianNB

In [43]:
gnb = GaussianNB()

In [44]:
gnb.fit(X_train , y_train)

GaussianNB()

In [45]:
y_pred = gnb.predict(X_test)

In [50]:
from sklearn.metrics import accuracy_score , precision_score , recall_score , f1_score 

In [51]:
# Calculate accuracy
accuracy = accuracy_score(y_test, y_pred)

# Calculate precision
precision = precision_score(y_test, y_pred)

# Calculate recall
recall = recall_score(y_test, y_pred)

# Calculate F1 score
f1 = f1_score(y_test, y_pred)


In [52]:
print("Accuracy:", accuracy)
print("Precision:", precision)
print("Recall:", recall)
print("F1 Score:", f1)

Accuracy: 0.8282608695652174
Precision: 0.7098039215686275
Recall: 0.9731182795698925
F1 Score: 0.8208616780045351


In [53]:
# cross val = 10 

In [54]:
from sklearn.model_selection import cross_val_score

In [55]:
gnb1 = GaussianNB()

In [56]:
gnb_scores = cross_val_score(gnb1 , X , y , cv = 10)

In [57]:
print(np.mean(gnb_scores))

0.8217391304347826


# BERNOLLIS NAIVES BAYES CLASSIFIER

In [58]:
from sklearn.naive_bayes import BernoulliNB

In [59]:
bnc = BernoulliNB()

In [63]:
bnc.fit(X_train , y_train)

BernoulliNB()

In [65]:
y_pred = bnc.predict(X_test)

In [67]:
# Calculate accuracy
accuracy = accuracy_score(y_test, y_pred)

# Calculate precision
precision = precision_score(y_test, y_pred)

# Calculate recall
recall = recall_score(y_test, y_pred)

# Calculate F1 score
f1 = f1_score(y_test, y_pred)


In [68]:
print("Accuracy:", accuracy)
print("Precision:", precision)
print("Recall:", recall)
print("F1 Score:", f1)

Accuracy: 0.8797101449275362
Precision: 0.8951612903225806
Recall: 0.7956989247311828
F1 Score: 0.842504743833017


In [62]:
bnc_scores = cross_val_score(bnc , X , y , cv = 10)

In [61]:
print(np.mean(bnc_scores))

0.8839130434782609


# MULTINOMIAL NAVES BAYES CLASSIFIER

In [69]:
from sklearn.naive_bayes import MultinomialNB

In [70]:
mnc = MultinomialNB()

In [71]:
mnc.fit(X_train , y_train)

MultinomialNB()

In [72]:
y_pred = mnc.predict(X_test)

In [73]:
# Calculate accuracy
accuracy = accuracy_score(y_test, y_pred)

# Calculate precision
precision = precision_score(y_test, y_pred)

# Calculate recall
recall = recall_score(y_test, y_pred)

# Calculate F1 score
f1 = f1_score(y_test, y_pred)


print("Accuracy:", accuracy)
print("Precision:", precision)
print("Recall:", recall)
print("F1 Score:", f1)

Accuracy: 0.8050724637681159
Precision: 0.7557522123893805
Recall: 0.7652329749103942
F1 Score: 0.7604630454140694


In [74]:
mnc_scores = cross_val_score(mnc , X , y , cv = 10)
print(np.mean(mnc_scores))

0.786086956521739


 # BERNOLLIS NAIVES BAYES CLASSIFIER performed best giving an 88 % of accuracy among all the other methods while Guassian and multinomial gave 82 and 78 % of mean scores respectively !